src/Cli/dotnet/Parser.cs

```diff
@@ -109,6 +109,7 @@ public static Command GetBuiltInCommand(string commandName)
 
         public static System.CommandLine.Parsing.Parser Instance { get; } = new CommandLineBuilder(ConfigureCommandLine(RootCommand))
             .UseExceptionHandler(ExceptionHandler)
+            .UseParseErrorReporting(127) // TODO:CH - what exit code for bad parsing?
```

Comment:
need to answer this question before merge

Reply:
this breaks parsing for commands that forward to msbuild.  @vlada-shubina I believe that I added this because you said we needed it for better parse errors, is that right? if so, we'll either need to:

* figure out a way to only apply this for `dotnet new` command invocations, or
* figure out ways for commands that forward another application to collect all of their arguments without error and forward them along to the underlying application

Reply:
quick example to verify that this is in fact causing issues:

```
dotnet msbuild /t:thingy
```

returns an exit code of 127 now

Reply:
as discussed, such arguments should be parsed with `TreatUnmatchedTokensAsErrors  = false`